pkg/spacebinding/spacebindinglister_test.go

```diff
@@ -235,4 +235,199 @@ func TestNewSpaceBindingLister(t *testing.T) {
+			spacetest.WithLabel(toolchainv1alpha1.SpaceCreatorLabelKey, "space-a-user"),
+		)
+		spaceB := spacetest.NewSpace(test.HostOperatorNs, "Space-B",
+			spacetest.WithTierName("advanced"),
+			spacetest.WithSpecTargetCluster(test.MemberClusterName),
+			spacetest.WithSpecParentSpace(spaceA.GetName()),
+			spacetest.WithDisableInheritance(true),
+		)
+		spaceC := spacetest.NewSpace(test.HostOperatorNs, "Space-C",
+			spacetest.WithTierName("advanced"),
+			spacetest.WithSpecTargetCluster(test.MemberClusterName),
+			spacetest.WithSpecParentSpace(spaceB.GetName()),
+			spacetest.WithDisableInheritance(false),
+		)
+		spaceD := spacetest.NewSpace(test.HostOperatorNs, "Space-D",
+			spacetest.WithTierName("advanced"),
+			spacetest.WithSpecTargetCluster(test.MemberClusterName),
+			spacetest.WithSpecParentSpace(spaceC.GetName()),
+			spacetest.WithDisableInheritance(false),
+		)
+		spaceE := spacetest.NewSpace(test.HostOperatorNs, "Space-E",
+			spacetest.WithTierName("advanced"),
+			spacetest.WithSpecTargetCluster(test.MemberClusterName),
+			spacetest.WithSpecParentSpace(spaceD.GetName()),
+			spacetest.WithDisableInheritance(true),
+		)
+		spaceF := spacetest.NewSpace(test.HostOperatorNs, "Space-F",
+			spacetest.WithTierName("advanced"),
+			spacetest.WithSpecTargetCluster(test.MemberClusterName),
+			spacetest.WithSpecParentSpace(spaceE.GetName()),
+			spacetest.WithDisableInheritance(false),
+		)
+		spaces := map[string]*toolchainv1alpha1.Space{
+			spaceA.GetName(): spaceA,
+			spaceB.GetName(): spaceB,
+			spaceC.GetName(): spaceC,
+			spaceD.GetName(): spaceD,
+			spaceE.GetName(): spaceE,
+			spaceF.GetName(): spaceF,
+		}
+		// the getSpaceFunc returns the space based on the given name
+		getSpaceFunc := func(spaceName string) (*toolchainv1alpha1.Space, error) {
+			if space, found := spaces[spaceName]; found {
+				return space, nil
+			}
+			return nil, fmt.Errorf("space not found: %s", spaceName)
+		}
+
+		// listSpaceBindingFunc returns the specific spacebindings for the given space
+		listSpaceBindingFunc := func(spaceName string) ([]toolchainv1alpha1.SpaceBinding, error) {
+			switch spaceName {
+			case spaceA.GetName():
+				return []toolchainv1alpha1.SpaceBinding{
+					*spacebinding.NewSpaceBinding(spaceAMur, spaceA, spaceA.Name, spacebinding.WithRole("admin")),
+				}, nil
+			case spaceB.GetName():
+				return []toolchainv1alpha1.SpaceBinding{
+					*spacebinding.NewSpaceBinding(spaceBMur, spaceB, spaceB.Name, spacebinding.WithRole("viewer")),
+				}, nil
+			case spaceC.GetName():
+				return []toolchainv1alpha1.SpaceBinding{
+					*spacebinding.NewSpaceBinding(spaceCMur, spaceC, spaceC.Name, spacebinding.WithRole("maintainer")),
+				}, nil
+			case spaceD.GetName():
+				return []toolchainv1alpha1.SpaceBinding{
+					*spacebinding.NewSpaceBinding(spaceDMur, spaceD, spaceD.Name, spacebinding.WithRole("admin")),
+				}, nil
+			case spaceE.GetName():fmt.Println("return no binding for SpaceE")
```

Comment:
even if I'm not sure we really need this print line. Maybe we could even remove it? 
```suggestion
			case spaceE.GetName():
			        fmt.Println("return no binding for SpaceE")
```